### DISCUSSION TITLE
How to run shell command from command palette?

### DISCUSSION THREAD

**Discussion by James-Firth** [helix-editor/helix#9580] [Q&A] [answered]
I just discovered the command palette (very cool and useful for learning commands!)

However when I select the `Run a shell command [:run-shell-command]` option with Enter it just performs that action and says "Command Succeeded".

My expectation is I would hit enter the `:run-shell-command` might be displayed as the selected command prefix and I could then type the shell command I wanted to run (eg: `git pull`) and hit enter again for it to run. (similar to how VS Code simply prompts me again for multi-step actions)

Is there a way for me to do this from the command palette or do I need to type `:sh`?
I'm fine with doing `:sh` it's just going to confuse anyone finding the command palette I suspect.

---

I searched for Discussions and Issues, and tried other keys such as tab, but nothing made sense to me.

**Comment by the-mikedavis** [comment, marked as answer]
`:sh` is just an alias for `:run-shell-command`. When we execute a typable command like `:run-shell-command` we execute it without arguments so the shell command is empty. I opened up an issue about changing the behavior to open the command mode prompt with the typable command prefilled instead: https://github.com/helix-editor/helix/issues/9584

**Reply by James-Firth** [reply]
Thanks Mike! That sounds like a great way to achieve what I was looking for
